### ISSUE TITLE
spawn EINVAL when building on windows on node 20.12+

### ISSUE THREAD

**Issue by zawata** [kriszyp/msgpackr-extract#13]
```
> msgpackr-extract@3.0.2 install
> node-gyp-build-optional-packages

node:internal/child_process:421
    throw new ErrnoException(err, 'spawn');
    ^

Error: spawn EINVAL
    at ChildProcess.spawn (node:internal/child_process:421:11)
    at Object.spawn (node:child_process:761:9)
    at build (C:\Users\john\Documents\msgpackr-extract\node_modules\node-gyp-build-optional-packages\bin.js:29:8)
    at preinstall (C:\Users\john\Documents\msgpackr-extract\node_modules\node-gyp-build-optional-packages\bin.js:38:32)
    at C:\Users\john\Documents\msgpackr-extract\node_modules\node-gyp-build-optional-packages\bin.js:11:7
    at ChildProcess.exithandler (node:child_process:430:5)
    at ChildProcess.emit (node:events:519:28)
    at maybeClose (node:internal/child_process:1105:16)
    at Socket.<anonymous> (node:internal/child_process:457:11)
    at Socket.emit (node:events:519:28) {
  errno: -4071,
  code: 'EINVAL',
  syscall: 'spawn'
}

Node.js v20.14.0
npm error code 1
```

This comes from `node-gyp-build-optional-packages`. I understand you [maintain a fork of node-gyp-build](https://github.com/kriszyp/node-gyp-build) while waiting for a PR to be merged upstream.

This error has already been fixed upstream: https://github.com/prebuild/node-gyp-build/pull/69 so the fix just needs to be backported or your changes rebased.

**Comment by kriszyp** [comment]
Should be addressed in v3.0.3, let me know if you still see an issue.

**Comment by zawata** [comment]
msgpackr-extract is an optional, transitive dependency like 3-4 levels deep in our project and I had to get some releases out so this took me some time to fully test.

This is fixed so I'll close the issue.

I've also been doing some testing with windows-on-arm64 and while this module doesn't have win32-arm64 prebuilts it does build correctly when running npm install. But when I build it as a depdendency of my app I'm getting an error that's killing the build process prematurely even though it's an optional dependency. I haven't determined where it's coming from though and it doesn't seem to be this one or msgpackr so I still need to track it down and will open a new issue if one of these module does become the culprit.

thanks!

EDIT:
this is the second issue I am encountering: https://github.com/prebuild/node-gyp-build/issues/71
As it isn't yet fixed upstream(and can be worked around by moving node to a path without spaces) I won't reopen this issue
